### PR TITLE
feat(payments): remove Coupons Feature Flag

### PR DESCRIPTION
**Pull request by IvoJP**
## Because

- We no longer need the ability to hide coupons

## This pull request

- Removes the feature flag that hides the coupon feature

## Issue that this pull request solves

Closes: #10853 

## Checklist

_Put an `x` in the boxes that apply_

- [X] My commit is GPG signed.
- [X] If applicable, I have modified or added tests which pass locally.
- [X] I have added necessary documentation (if appropriate).
- [X] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

**Comment by biancadanforth** [comment]
@IvoJP : Thanks for the PR; it'd probably be a good idea to make sure QA can verify an active/valid coupon in prod before removing the flag. What do you think?

File: packages/fxa-auth-server/config/index.ts
Comment (biancadanforth) [inline packages/fxa-auth-server/config/index.ts:867]: Huh; I guess this was never used anywhere after it was added in #11048 . 😅 At least we separately blocked coupons in the front-end via the other flag.

**Review by biancadanforth — APPROVED** [review]
LGTM, but I don't think we should merge this until QA has verified coupons in prod. CCing @annasob in case she feels differently.

Also, as a follow-up, we may want to remove these flags from the ops repos (e.g. [payments flag](https://github.com/mozilla-services/cloudops-infra/blob/7b1e5fcaa459025eb1a3102596756264c936ec8f/projects/fxa/k8s/values/payments.yaml#L12-L13)) as well? We could remove the PayPal feature flag(s) while we're in there.

**Comment by StaberindeZA** [comment]
@annasob should we wait for confirmation in Production before merging this, or could we merge this PR?

**Comment by annasob** [comment]
Chelsea will talk to product about making a 1% off coupon for testing. Lets
wait on this.

**Comment by bbangert** [comment]
@IvoJP Sounds like we're good for this to land, just needs a rebase for conflicts!

**Pull request merged**
